### PR TITLE
Add proxy protocol

### LINKED ISSUE #384: Proxy Protocol Support
Is there any plans for support for [PROXY protocol](http://www.haproxy.org/download/1.5/doc/proxy-protocol.txt)?

### PR DESCRIPTION
**Pull request by emilevauge**
### Description

This PR supersedes #1145 to add [Proxy Protocol](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) support. It is based on the initial work made by [jrb](https://github.com/jrb).
it also bumps Golang to 1.9 (needed to add Proxy Protocol).

Fixes #384

### CODE REVIEW

File: build.Dockerfile [hunk 1]
```diff
@@ -1,11 +1,8 @@
-FROM golang:1.8
+FROM golang:1.9-alpine
```
Comment (timoreimann) [inline build.Dockerfile]: Do we plan to merge this even when Go 1.9 is still beta? I'd be concerned about that.

Comment (emilevauge) [inline build.Dockerfile]: We think the go 1.9 final will be out way before the trafik 1.4 final is ready, and this feature is the most wanted feature ;) So all in all, we think it's a reasonable risk.

Comment (timoreimann) [inline build.Dockerfile]: What gives you confidence the next Go release will be out way before Traefik 1.4? Go 1.9 is only at the first release candidate, and Traefik is going to ship its first within the next few days presumably.

Comment (emilevauge) [inline build.Dockerfile]: @timoreimann the rc2 was released weeks ago, and the final release date was theoretically fixed weeks ago, and also https://github.com/golang/go/milestone/49 ;)

Comment (timoreimann) [inline build.Dockerfile]: OK, missed that it's actually rc2 already.

I'm still not sure it's a good idea to merge this before Go 1.9 final is actually released. I have seen things break in subtle ways over at least the last two Go releases. I'd prefer to take the time after the code freeze to iron out potential problems.

File: server/configuration.go [hunk 2]
```diff
@@ -247,12 +247,18 @@ func (ep *EntryPoints) Set(value string) error {
 		whiteListSourceRange = strings.Split(result["WhiteListSourceRange"], ",")
 	}
 
+	proxyprotocol := false
+	if len(result["ProxyProtocol"]) > 0 {
+		proxyprotocol = strings.EqualFold(result["ProxyProtocol"], "enable") || strings.EqualFold(result["ProxyProtocol"], "on")
```
Comment (dtomcej) [inline server/configuration.go]: You check to see if `proxyprotocol` is `enable` or `on`, but not `true` based on your example in the toml?

Comment (emilevauge) [reply server/configuration.go]: Yeah, I just copied what was already done for `Compress` flag. Added `true for both :)

File: build.Dockerfile (hunk 1 above)
Comment (racastellanosm) [reply build.Dockerfile]: I think a lot of people need the Proxy Protocol Support (included myself)! Any release date in the scope?

**Pull request merged**